src/DssSpell.t.sol

```diff
@@ -354,33 +373,77 @@ contract DssSpellTest is DssSpellTestBase {
         // assertEq(dai.balanceOf(wallets.addr("DAIF_WALLET")), prevBalance + 67_863 * WAD);
     }
 
-    function testFlash() public {
+    function _setupRootDomain() internal {
+        vm.makePersistent(address(spell), address(spell.action()));
+
+        string memory root = string.concat(vm.projectRoot(), "/lib/dss-test");
+        config = ScriptTools.readInput(root, "integration");
+
+        rootDomain = new RootDomain(config, getRelativeChain("mainnet"));
+    }
+
+    function testL2OptimismSpell() public {
+        _setupRootDomain();
+        
+        optimismDomain = new OptimismDomain(config, getRelativeChain("optimism"), rootDomain);
+        optimismDomain.selectFork();
+
+        // Check that the L2 Optimism Spell is there and configured
+        L2Spell optimismSpell = L2Spell(0xC077Eb64285b40C86B40769e99Eb1E61d682a6B4);
 
-        FlashAbstract flashLegacy = FlashAbstract(addr.addr("MCD_FLASH_LEGACY"));
-        FlashAbstract flashCurrent = FlashAbstract(addr.addr("MCD_FLASH"));
+        L2Gateway optimismGateway = L2Gateway(optimismSpell.gateway());
+        assertEq(address(optimismGateway), 0xd9e000C419F3aA4EA1C519497f5aF249b496a00f, "l2-optimism-wrong-gateway");
```

Comment:
Instead of comparing to the hard-coded value this address can be pre-read in l1 from OPTIMISM_TELEPORT_BRIDGE.l2TeleportGateway.
Same for Arbitrum below.

Reply:
updated